docs/deployment/docker.md

```diff
@@ -58,24 +58,30 @@ Once the host build is complete, we are ready to build our image. The following
 ```Dockerfile
 FROM node:16-bullseye-slim
 
-WORKDIR /app
-
-# install sqlite3 dependencies, you can skip this if you don't use sqlite3 in the image
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
     rm -rf /var/lib/apt/lists/* && \
     yarn config set python /usr/bin/python3
 
+# From here on we use the least-privileged `node` user to run the backend.
+USER node
+WORKDIR /app
+
+# This switches many Node.js dependencies to production mode.
+ENV NODE_ENV production
+
 # Copy repo skeleton first, to avoid unnecessary docker cache invalidation.
 # The skeleton contains the package.json of each package in the monorepo,
 # and along with yarn.lock and the root package.json, that's enough to run yarn install.
-COPY yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
+COPY --chown=node:node yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
 RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
```

Comment:
do we need to chown after unpacking? doesn't tar preserve uid/gid from within the tar file by default? Same in the other usages of tar below

Reply:
Nop that all seems fine:

<img width="497" alt="image" src="https://user-images.githubusercontent.com/4984472/183903651-29a7857a-2300-48af-a4ad-a09fba6a34aa.png">

I believe it's because the CLI creates a "portable" tar, and we're unpacking as the node user

Reply:
@freben @Rugvip You would need to run `--chown node:node` for `app-config*.yaml` and any other files that you copy in after the tars. For example, with the following:

```Dockerfile
COPY packages/backend/dist/bundle.tar.gz ./
RUN tar xzf bundle.tar.gz && rm bundle.tar.gz

COPY app-config*.yaml packages/backend/docker-entrypoint.sh ./
COPY packages/catalog-model ./packages/catalog-model
```

You have the following permissions:

```shell
/app $ ls -lAF
total 912
-rw-r--r--    1 root     root          2309 Aug 15 17:12 app-config.yaml
-rwxr-xr-x    1 root     root           934 Aug 11 23:21 docker-entrypoint.sh*
drwxr-xr-x  614 node     node         20480 Aug 15 17:34 node_modules/
-rw-r--r--    1 node     node          1692 Jan  1  1970 package.json
drwxr-xr-x    1 node     node          4096 Aug 15 17:35 packages/
-rw-r--r--    1 node     node        889052 Jan  1  1970 yarn.lock

/app $ ls -lAF packages
total 16
drwxr-xr-x    1 node     node          4096 Aug 15 17:34 app/
drwxr-xr-x    1 node     node          4096 Aug 15 17:34 backend/
drwxr-xr-x    3 root     root          4096 Aug 15 17:35 catalog-model/
```

Reply:
This change adds the `--chown node:node` to the copy instructions.

Regarding adding files to the bundle, any file that is referenced from `"files"` in `package.json`s will be included in the bundle.

Reply:
@Rugvip Yep yep, sorry, I was more commenting on the fact that anything outside of a tar will need a `--chown` flag. Might be something good for the documentation to alleviate support requests.